src/components/DistributionAccountCircle.tsx

```diff
@@ -0,0 +1,294 @@
+import { Box } from "components/Box";
+import { LoadingContent } from "components/LoadingContent";
+import { AssetAmount } from "components/AssetAmount";
+import { DropdownMenu } from "components/DropdownMenu";
+import { MoreMenuButton } from "components/MoreMenuButton";
+import { NotificationWithButtons } from "components/NotificationWithButtons";
+
+import { AppDispatch } from "store";
+import { getOrgCircleInfoAction } from "store/ducks/organization";
+import { useRedux } from "hooks/useRedux";
+import { useCircleBalances } from "apiQueries/useCircleBalances";
+import { useUpdateCircleConfig } from "apiQueries/useUpdateCircleConfig";
+
+export const DistributionAccountCircle = () => {
+  const CIRCLE_API_MASKED_VALUE = "xxxx";
+
+  const { organization } = useRedux("organization");
+  const { distributionAccount } = organization.data;
+  const isPendingAccount =
+    distributionAccount?.status === "PENDING_USER_ACTIVATION";
+
+  const [isEditMode, setIsEditMode] = useState(isPendingAccount);
+  const [circleApiKey, setCircleApiKey] = useState("");
+  const [walletId, setWalletId] = useState("");
+  const [configResponse, setConfigResponse] = useState<{
+    message: string;
+    type: "error" | "success";
+  } | null>(null);
+
+  const {
+    data: balances,
+    error: balanceError,
+    isFetching: isBalancesFetching,
+    isLoading: isBalancesLoading,
+  } = useCircleBalances(Boolean(distributionAccount?.circleWalletId));
+
+  const {
+    data: circleConfig,
+    isFetching: isCircleConfigFetching,
+    isError: isCircleConfigError,
+    isSuccess: isCircleConfigSuccess,
+    error: circleConfigError,
+    refetch: updateCircleConfig,
+  } = useUpdateCircleConfig({ api_key: circleApiKey, wallet_id: walletId });
+
+  const queryClient = useQueryClient();
+  const dispatch: AppDispatch = useDispatch();
+
+  useEffect(() => {
+    if (isEditMode && (isCircleConfigSuccess || isCircleConfigError)) {
+      setConfigResponse({
+        message: isCircleConfigSuccess
+          ? circleConfig.message
+          : circleConfigError.message,
+        type: isCircleConfigSuccess ? "success" : "error",
+      });
+
+      // Reset query
+      queryClient.resetQueries({
+        queryKey: ["organization", "circle", "config"],
+      });
+
+      // Fetch new data
+      if (isCircleConfigSuccess) {
+        dispatch(getOrgCircleInfoAction());
+        setIsEditMode(false);
+      }
+    }
+    // Not including circleConfig.message and circleConfigError.message
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isCircleConfigSuccess, isCircleConfigError, queryClient, dispatch]);
+
+  const isSaveEnabled = Boolean(
+    (circleApiKey && circleApiKey !== CIRCLE_API_MASKED_VALUE) ||
+      (walletId && walletId !== distributionAccount?.circleWalletId),
+  );
+
+  const handleUpdate = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    updateCircleConfig();
+  };
```

Comment:
👋  As I mentioned in [this thread](https://stellarfoundation.slack.com/archives/C04C9MLM9UZ/p1719355360126979?thread_ts=1719234904.950069&cid=C04C9MLM9UZ), there's a small bug where the circle balances should be re-fetched after the circle config is updated.

Reply:
Thanks for catching this! :raised_hands: